python/tvm/micro/base.py

```diff
@@ -0,0 +1,190 @@
+
+
+def cross_compiler(toolchain_prefix, include_dev_lib_header=True):
+    """Creates a cross compile function that wraps `create_micro_lib`.
+
+    For use in `tvm.module.Module.export_library`.
+
+    Parameters
+    ----------
+    toolchain_prefix : str
+        toolchain prefix to be used
+
+    include_dev_lib_header : Optional[bool]
+        whether to include the device library header containing definitions of
+        library functions.
+
+    Return
+    ------
+    func : Callable[[str, str, Optional[str]], None]
+        cross compile function taking a destination path for the object file
+        and a path for the input source file.
+
+    Example
+    --------
+    .. code-block:: python
+
+      c_mod = ...  # some module generated with "c" as the target
+      fcompile = tvm.micro.cross_compiler(toolchain_prefix="")
+      c_mod.export_library("dev_lib.obj", fcompile=fcompile)
+    """
+    def compile_func(obj_path, src_path, **kwargs):
+        if isinstance(obj_path, list):
+            obj_path = obj_path[0]
+        if isinstance(src_path, list):
+            src_path = src_path[0]
+        create_micro_lib(obj_path, src_path, toolchain_prefix,
+                         kwargs.get("options", None), include_dev_lib_header)
+    return _cc.cross_compiler(compile_func)
+
+
+def create_micro_lib(
+        obj_path, src_path, toolchain_prefix, options=None, include_dev_lib_header=True):
+    """Compiles code into a binary for the target micro device.
+
+    Parameters
+    ----------
+    obj_path : Optional[str]
+        path to generated object file (defaults to same directory as `src_path`)
+
+    src_path : str
+        path to source file
+
+    toolchain_prefix : str
+        toolchain prefix to be used
+
+    include_dev_lib_header : bool
+        whether to include the device library header containing definitions of
+        library functions.
+    """
+    def replace_suffix(s, new_suffix):
+        if "." in os.path.basename(s):
+            # There already exists an extension.
+            return os.path.join(
+                os.path.dirname(s),
+                ".".join(os.path.basename(s).split(".")[:-1] + [new_suffix]))
+        # No existing extension; we can just append.
+        return s + "." + new_suffix
+
+    # uTVM object files cannot have an ".o" suffix, because it triggers the
+    # code path for creating shared objects in `tvm.module.load`.  So we replace
+    # ".o" suffixes with ".obj".
+    if obj_path.endswith(".o"):
+        logging.warning(
+            "\".o\" suffix in \"%s\" has been replaced with \".obj\"", obj_path)
+        obj_path = replace_suffix(obj_path, "obj")
+
+    options = ["-I" + path for path in find_include_path()]
+    options += ["-I{}".format(_get_micro_device_dir())]
+    options += ["-fno-stack-protector"]
+    # TODO(weberlo): This option cannot be used on 32-bit machines. Make this
+    # compilation pipeline compatible with 32-bit.
+    options += ["-mcmodel=large"]
```

Comment:
This option should be host based . Usually the '-m' prefixed options are machine specific and hard coded. While -mcmodel=large would likely work on other hosts, we should aim to keep this largely host clean . 

Maybe a check that this is an x86_64-linux host might be appropriate here especially as you are looking to create a large model .so ?